Userland/Libraries/LibWeb/HTML/ImageData.cpp

```diff
@@ -37,6 +39,50 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<ImageData>> ImageData::construct_impl(JS::R
     return ImageData::create(realm, sw, sh, settings);
 }
 
+// https://html.spec.whatwg.org/multipage/canvas.html#dom-imagedata-with-data
+WebIDL::ExceptionOr<JS::NonnullGCPtr<ImageData>> ImageData::create(JS::Realm& realm, JS::Handle<WebIDL::BufferSource> const& data, u32 sw, Optional<u32> sh, Optional<ImageDataSettings> const&)
+{
+    auto& vm = realm.vm();
+
+    if (!is<JS::Uint8ClampedArray>(*data->raw_object()))
+        return vm.throw_completion<JS::TypeError>(JS::ErrorType::NotAnObjectOfType, "Uint8ClampedArray");
+
+    auto& uint8_clamped_array_data = static_cast<JS::Uint8ClampedArray&>(*data->raw_object());
+
+    // 1. Let length be the number of bytes in data.
+    auto length = uint8_clamped_array_data.byte_length().length();
+
+    // 2. If length is not a nonzero integral multiple of four, then throw an "InvalidStateError" DOMException.
+    if (length % 4 != 0)
```

Comment:
I didn't notice this on first read and thought the NOTE on step 4 didn't make sense, but this doesn't quite match the text as it's missing the `nonzero` part, the integral multiple of four is correct though.

Reply:
Great catch!